From: Julio Ramos <ramos.isw@gmail.com>
Subject: [PATCH] sunxi: add PBS TV Stick DT and defconfig

Add arch/arm/dts/sun50i-h5-pbstvstick.dts and configs/pbstvstick_defconfig
for PBS TV Stick (PBST314).

---
 arch/arm/dts/sun50i-h5-pbstvstick.dts | 59 +++++++++++++++++++++++++++++++++++
 configs/pbstvstick_defconfig          | 21 +++++++++++
 2 files changed, 80 insertions(+)

diff --git a/arch/arm/dts/sun50i-h5-pbstvstick.dts b/arch/arm/dts/sun50i-h5-pbstvstick.dts
new file mode 100644
--- /dev/null
+++ b/arch/arm/dts/sun50i-h5-pbstvstick.dts
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2026
+ */
+
+/dts-v1/;
+#include "sun50i-h5-orangepi-prime.dts"
+
+/ {
+	model = "PBST314";
+	compatible = "shaghal,pbstvstick", "allwinner,sun50i-h5";
+
+	leds {
+		led-0 {
+			label = "pbstvstick:blue:pwr";
+			gpios = <&pio 0 15 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+	};
+};
+
+/delete-node/ &wifi_pwrseq;
+
+&reg_vcc3v3 {
+	u-boot,dm-spl;
+};
+
+&mmc0_pins {
+	u-boot,dm-spl;
+};
+
+&mmc0 {
+	u-boot,dm-spl;
+	vmmc-supply = <&reg_vcc3v3>;
+	bus-width = <4>;
+	cap-sd-highspeed;
+	broken-cd;
+	disable-wp;
+	no-1-8-v;
+	status = "okay";
+};
+
+&mmc1 {
+	status = "disabled";
+	/delete-property/ mmc-pwrseq;
+};
+
+&mmc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&mmc2_8bit_pins>;
+	vmmc-supply = <&reg_vcc3v3>;
+	bus-width = <4>;
+	max-frequency = <25000000>;
+	vqmmc-supply = <&reg_vcc3v3>;
+	no-1-8-v;
+	non-removable;
+	cap-mmc-hw-reset;
+	status = "okay";
+};
diff --git a/configs/pbstvstick_defconfig b/configs/pbstvstick_defconfig
new file mode 100644
--- /dev/null
+++ b/configs/pbstvstick_defconfig
@@ -0,0 +1,21 @@
+CONFIG_ARM=y
+CONFIG_ARCH_SUNXI=y
+CONFIG_DEFAULT_DEVICE_TREE="sun50i-h5-pbstvstick"
+# CONFIG_OF_UPSTREAM is not set
+CONFIG_DRAM_CLK=432
+CONFIG_SPL=y
+CONFIG_MACH_SUN50I_H5=y
+CONFIG_DRAM_ZQ=3881977
+CONFIG_DRAM_ODT_EN=y
+CONFIG_MMC_SUNXI_SLOT_EXTRA=2
+CONFIG_SPL_MMC_TINY=y
+# CONFIG_SPL_DM_MMC is not set
+# CONFIG_SPL_SPI_SUNXI is not set
+# CONFIG_SUPPORT_EMMC_BOOT is not set
+# CONFIG_SPL_WDT is not set
+# CONFIG_WDT is not set
+# CONFIG_WATCHDOG is not set
+CONFIG_SUN8I_EMAC=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_OHCI_HCD=y
+CONFIG_USB_MUSB_GADGET=y
